internal/internal_task_handlers.go

```diff
@@ -367,9 +367,26 @@ OrderEvents:
 	eh.loadedEvents = eh.loadedEvents[eh.currentIndex:]
 	eh.currentIndex = 0
 
+	// calculate estimated size of all events, for workflow info
+	historySize := estimatedSize(nextEvents) + estimatedSize(markers)
+
 	return nextEvents, markers, nil
 }
 
+func estimatedSize(events []s.HistoryEvent) int {
+	sum := 0
+	for _, e := range events {
+		switch *e.EventType {
+		case s.EventTypeWorkflowExecutionStarted:
+			sum += len(e.WorkflowExecutionStartedEventAttributes.Input)
+		// ...
+		case s.EventTypeWorkflowExecutionSignaled:
+			sum += len(e.WorkflowExecutionSignaledEventAttributes.Input)
+		}
+	}
+	return sum
+}
```

Comment:
basic thought here: if we add fields in the future (to either client or server) we could change the calculation if we counted _the whole event_, so don't do that.  count only "big" user-controlled fields, and assume that's close enough (maybe with an offset to get a more accurate estimate).

so stuff like:
- input
- headers
- reset points (maybe) (because they can be large)
- search attributes
- etc

but not e.g. workflow ID fields or anything that's "small" or could be migrated to a different field / different lengths in the future.

server-side the calculation for size limits is simply the whole list of events serialized as thrift.  the server can change this arbitrarily and reject new events without causing determinism failures, but we can't do that safely here because adding any RPC field could change the calculation.

---

alternatively, we could calculate _the size we originally told the client_ and store that in the events, which would allow us to return arbitrary data and claim it's a different size.  that would get us more accuracy + strong stability at the cost of a small bit more storage.

I'm not sure that's worth it though, and it still can't be _truly_ precise because it has to be backwards compatible at all times, and making it intelligent about client versions / changing calculations / etc may be too complex anyway.  Just counting the "big" things seems fair enough.